tests/util/sampler.py

```diff
@@ -0,0 +1,41 @@
+from abc import ABC
+
+import tensorflow as tf
+
+from trieste.type import TensorType
+from trieste.utils import DEFAULTS
+
+
+class PseudoBatchReparametrizationSampler(ABC):
+    """A Sampler that takes the specified sample as deterministic samples`."""
+
+    def __init__(self, samples: TensorType):
+        """
+        :param samples `[S, B, L]`, where `S` is the `sample_size`, `B` the
+            number of points per batch, and `L` the dimension of the model's predictive
+            distribution.
+        """
+        tf.debugging.assert_shapes(
+            [(samples, ["S", "B", "L"])],
+            message="This acquisition function only supports batch sizes of one.",
+        )
+        self.samples = samples  # [S, B, L]
+
+    def sample(self, at: TensorType, *, jitter: float = DEFAULTS.JITTER) -> TensorType:
+        """
+        :param at: Batches of query points at which to sample the predictive distribution, with
+            shape `[..., B, D]`, for batches of size `B` of points of dimension `D`.
+        :param jitter: placeholder
+        :return: The samples, of shape `[..., S, B, L]`, where `S` is the `sample_size`, `B` the
+            number of points per batch, and `L` the dimension of the model's predictive
+            distribution.
+        """
+
+        tf.debugging.assert_rank_at_least(at, 2)
+        tf.debugging.assert_greater_equal(jitter, 0.0)
+
+        batch_size = at.shape[-2]
```

Comment:
Why do we have this?

Reply:
It is aimed for testing, in the [referred code](https://github.com/pytorch/botorch/blob/master/test/acquisition/multi_objective/test_monte_carlo.py), they test the correctness of the utility function through specified mc samples.

This class is hence given as one may specify mc samples in this sampler and this will just return those fixed samples for acquisition function calculation, which can be used as a verification whether utility & acquisition function returns expected values.

Reply:
What do you think of this @uri-granta ?